File: 4-strings/4-mutable-vs-immutable/main.java
// Mutable vs Immutable Strings - Java Example

public class MutableVsImmutable {
    public static void main(String[] args) {
        String text = "hello";
        System.out.println("Original text: " + text);

        // String is immutable, so replace returns a new string
        String newText = text.replace('h', 'j');
        System.out.println("New text after replace: " + newText);
        System.out.println("Original text remains: " + text);

        // Using StringBuilder for mutable strings
        StringBuilder builder = new StringBuilder("hello");
        builder.setCharAt(0, 'j');
        System.out.println("Mutable StringBuilder text: " + builder.toString());
    }
}

File: 4-strings/4-mutable-vs-immutable/main.cpp
// Mutable vs Immutable Strings - C++ Example

#include <iostream>
#include <string>

int main() {
    std::string text = "hello";
    std::cout << "Original text: " << text << std::endl;

    text[0] = 'j';  // Mutable: changing first character
    std::cout << "Modified text: " << text << std::endl;

    return 0;
}
